nix/default.nix

```diff
@@ -0,0 +1,48 @@
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, sourcesOverride ? {}
+}:
+let
+  sources = import ./sources.nix { inherit pkgs; }
+    // sourcesOverride;
+  iohKNix = import sources.iohk-nix {};
+  haskellNix = import sources."haskell.nix";
+  # use our own nixpkgs if it exists in our sources,
+  # otherwise use iohkNix default nixpkgs.
+  nixpkgs = if (sources ? nixpkgs)
+    then (builtins.trace "Not using IOHK default nixpkgs (use 'niv drop nixpkgs' to use default for better sharing)"
+      sources.nixpkgs)
+    else (builtins.trace "Using IOHK default nixpkgs"
+      iohKNix.nixpkgs);
+
+  # for inclusion in pkgs:
+  overlays =
+    # Haskell.nix (https://github.com/input-output-hk/haskell.nix)
+    haskellNix.overlays
+    # haskell-nix.haskellLib.extra: some useful extra utility functions for haskell.nix
+    ++ iohKNix.overlays.haskell-nix-extra
+    # iohkNix: nix utilities and niv:
+    ++ iohKNix.overlays.iohkNix
+    # our own overlays:
+    ++ [
+      (pkgs: _: with pkgs; {
+
+        # commonLib: mix pkgs.lib with iohk-nix utils and our own:
+        commonLib = lib // iohkNix
+          // import ./util.nix { inherit lib haskell-nix; }
+          # also expose our sources and overlays
+          // { inherit overlays sources; };
+      })
+      # And, of course, our haskell-nix-ified cabal project:
+      (import ./pkgs.nix)
+      (import ./overlays/musl.nix)
```

Comment:
This should only be in the static musl version, but it probably won't cause any problems here.

Reply:
we *could* guard it, by adding additional `super.stdenv.hostPlatform.isMusl` guards. I don't think you would really want multiple nixpkgs instances at various places :-/

Reply:
Fair.